appserver/node-express/lib/db-client/qnaDoc/patch.js

```diff
@@ -66,6 +116,24 @@ module.exports = function (txid, spec) {
     );
   };
 
+  /**
+   * Handle the post of a new answer to a Samplestack question. The posted
+   * answer is merged with an answer template object, a unique ID, contributor
+   * information, and the current time. The resulting object is written to the
+   * MarkLogic database. During the write, the answer count for the question is
+   * incremented and the last activity date is updated.
+   *
+   * @param  {String} txid The transaction ID.
+   * @param  {Object} contributor Contributor object. Example:
+   *   {"id":"cf99542d-f024-4478-a6dc-7e723a51b040",
+   *    "displayName":"JoeUser"}
+   * @param  {String} questionId The question ID.
+   * @param  {Object} spec The answer content from the request. Example:
+   *   {"comments":[],
```

Comment:
Actually I've at least partially reworked the "draft" stuff. If I left it such that it's posting empty arrays, then so be it. I don't know I fully understand how this array came to be of interest. 

- If it's coming from the browser into the middle tier, then by all means the middle tier should dutifully store it, since that property being an empty array is entirely valid. 
- If the middle tier is generating it "for no particular reason," and both the Java and Node tiers work just fine without it being there at all, then it probably shouldn't be generating it all on its own.
- If the middle tier is generating it because it's necessary (for the indexing code, for instance), then it should be generated by the dbClient code and not the router/app logic code, as it would be a "phyiscal data need" rather than a logical need to express the lack of comments.

Does that make sense?